mdp/nodes/hsfa_nodes.py

```diff
@@ -0,0 +1,540 @@
+import os
+import tempfile
+import warnings as _warnings
+import webbrowser
+import mdp
+
+
+class HSFANode(mdp.Node):
+    """Extract slowly varying components via hierarchical
+    processing from input data. The network uses weight sharing for
+    each layer over multiple locally connected receptive fields.
+
+    More information about Hierarchical Slow Feature Analysis can be found in
+    Wiskott, L. and Sejnowski, T.J., Slow Feature Analysis: Unsupervised
+    Learning of Invariances, Neural Computation, 14(4):715-770 (2002).
+
+    HSFANode also supports random sampling of data-cuboids via a RandomSwitchboard
+    instead of the fixed RectangularSwitchboard.
+
+    HSFANode also implements most of the 'list' container methods; such as insert,
+    add, delete layers from the network.
+
+    The node also provides a 'show_flow' utility method to visualize the
+    network.
+
```

Comment:
What is the advantage of this class over the FlowNode approach?
I'm also worried that this class includes too much different functionality (e.g., that it has a show_flow method). It sounds a bit like a god class.
I'd probably rather provide a helper function to create HSFA networks.